packages/devtools_app/lib/src/screens/inspector_v2/layout_explorer/box/box.dart

```diff
@@ -104,63 +103,110 @@ class BoxLayoutExplorerWidgetState extends LayoutExplorerWidgetState<
+        labelColor: widgetColor,
+        topPadding: hasTopPadding,
+        bottomPadding: hasBottomPadding,
+        leftPadding: hasLeftPadding,
+        rightPadding: hasRightPadding,
+      ),
+      if (hasLeftPadding)
+        PaddingVisualizerWidget(
+          RenderProperties(
+            axis: Axis.horizontal,
+            size: Size(displayLeftPadding, displayWidgetHeight),
+            offset: Offset(0, displayTopPadding),
+            realSize: Size(leftPadding, parentHeight),
+            layoutProperties: childProperties,
+            isFreeSpace: true,
+          ),
+          horizontal: true,
+        ),
+      if (hasTopPadding)
+        PaddingVisualizerWidget(
+          RenderProperties(
+            axis: Axis.horizontal,
+            size: Size(displayWidgetWidth, displayTopPadding),
+            offset: Offset(displayLeftPadding, 0),
+            realSize: Size(parentWidth, topPadding),
+            layoutProperties: childProperties,
+            isFreeSpace: true,
+          ),
+          horizontal: false,
+        ),
+      if (hasRightPadding)
+        PaddingVisualizerWidget(
+          RenderProperties(
+            axis: Axis.horizontal,
+            size: Size(displayRightPadding, displayWidgetHeight),
+            offset: Offset(
+              displayLeftPadding + displayWidgetWidth,
+              displayTopPadding,
+            ),
+            realSize: Size(rightPadding, parentHeight),
+            layoutProperties: childProperties,
+            isFreeSpace: true,
+          ),
+          horizontal: true,
+        ),
+      if (hasBottomPadding)
+        PaddingVisualizerWidget(
+          RenderProperties(
+            axis: Axis.horizontal,
+            size: Size(displayWidgetWidth, displayBottomPadding),
+            offset: Offset(
+              displayLeftPadding,
+              displayTopPadding + displayWidgetHeight,
+            ),
+            realSize: Size(parentWidth, bottomPadding),
+            layoutProperties: childProperties,
+            isFreeSpace: true,
+          ),
+          horizontal: false,
+        ),
+    ];
+  }
 
-    double totalFraction = 0.0;
-    for (int i = 0; i < length; i++) {
-      final size = sizes[i];
-      if (size != null) {
-        fractions[i] = math.max(size / total, minFractions[i]);
-        totalFraction += fractions[i];
-      } else {
-        fractions[i] = 0.0;
-      }
-    }
-    if (totalFraction != 1.0) {
-      for (int i = 0; i < length; i++) {
-        fractions[i] = fractions[i] / totalFraction;
-      }
-    }
-    final output = <double>[];
-    for (final fraction in fractions) {
-      output.add(fraction * availableSize);
-    }
-    return output;
+  LayoutProperties? get parentProperties {
+    final parentElement = properties?.node.parentRenderElement;
+    if (parentElement == null) return null;
+    final parentProperties = computeLayoutProperties(parentElement);
+    return parentProperties;
```

Comment:
this can just be `return computeLayoutProperties(parentElement);`

Reply:
Done!